src/core/core-varlink.c

```diff
@@ -202,11 +202,24 @@ static int vl_method_subscribe_managed_oom_cgroups(
                 void *userdata) {
 
         _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        _cleanup_free_ char *username = NULL;
         Manager *m = ASSERT_PTR(userdata);
+        uid_t uid;
         int r;
 
         assert(link);
 
+        r = varlink_get_peer_uid(link, &uid);
```

Comment:
Hmm, I thought we had a rule that PID1 was not allowed to do nss lookups because that is potentially blocking. But now I'm confused, because `chase_symlinks()` calls `uid_to_name()`, added in f1bda7ead20c210a28be5decad636e1d95d8c3ec. Does it need to be reverted? @poettering PTAL.

Reply:
I guess that makes sense. But I see in this comment https://github.com/systemd/systemd/blob/c76691d708ac7fe13b7c4307c010d447fcbc3e9a/src/shared/specifier.c#L382-L389
that if UID is 0 it won't result in NSS. Which makes this okay?

Reply:
I opened https://github.com/systemd/systemd/pull/24934 for the uid_to_name() uses in other places.

The use in `specifier_user_name()` is OK because we know that `uid==0` if we're running as PID1, and that is short-circuited to return "root". But here we're explicitly querying `uid`, so it can be anything and an nss lookup might occur.

Reply:
Yeah, I'm pretty sure we can't do the lookup here.

Reply:
> The use in `specifier_user_name()` is OK because we know that `uid==0` if we're running as PID1, and that is short-circuited to return "root". But here we're explicitly querying `uid`, so it can be anything and an nss lookup might occur.

Oh I see I didn't read that right. Ok I'll rework this commit and try something different.